File: Plugins/Aspose_Words_Java_for_Ruby/lib/asposewordsjavaforruby/loadandsavetostream.rb
module Asposewordsjavaforruby
  module LoadAndSaveToStream
    def initialize()
        # Load and save to stream.
        save_to_stream()
    end
        
    def save_to_stream()
        # The path to the documents directory.
        data_dir = File.dirname(File.dirname(File.dirname(__FILE__))) + '/data/quickstart/'

        # Open the stream. Read only access is enough for Aspose.Words to load a document.
        stream = Rjb::import('java.io.FileInputStream').new(data_dir + "Document.doc")

        # Load the entire document into memory.
        doc = Rjb::import('com.aspose.words.Document').new(stream)

        # You can close the stream now, it is no longer needed because the document is in memory.
        stream.close()
        # ... do something with the document
        # Convert the document to a different format and save to stream.
        dst_stream = Rjb::import("java.io.ByteArrayOutputStream").new()
        save_format = Rjb::import("com.aspose.words.SaveFormat")
        doc.save(dst_stream, save_format.RTF)

        output = Rjb::import("java.io.FileOutputStream").new(data_dir + "Document Out.rtf")
        output.write(dst_stream.toByteArray())
        output.close()
    end 

  end
end

File: Examples/src/main/java/com/aspose/words/examples/loading_saving/LoadAndSaveToStream.java
package com.aspose.words.examples.loading_saving;

import com.aspose.words.Document;
import com.aspose.words.SaveFormat;
import com.aspose.words.examples.Utils;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.InputStream;
import java.io.OutputStream;
import java.sql.Connection;

public class LoadAndSaveToStream {
	private static Connection mConnection;

	public static void main(String[] args) throws Exception {

		// ExStart:LoadAndSaveToStream
		// The path to the documents directory.
		String dataDir = Utils.getDataDir(LoadAndSaveToStream.class);
		String inputFile = "Test File (doc).doc";
		String outputFile = "output.png";

		InputStream in = new FileInputStream(dataDir + inputFile);
		OutputStream out = new FileOutputStream(dataDir + outputFile);

		Document doc = new Document(in);

		// Save the finished document to disk.
		doc.save(out, SaveFormat.PNG);
		// ExEnd:LoadAndSaveToStream
		System.out.println("Document loaded and saved successfully.");
	}
}
